opal/util/keyval_parse.c

```diff
@@ -200,53 +201,58 @@ int opal_util_keyval_save_internal_envars(opal_keyval_parse_fn_t callback)
     return OPAL_SUCCESS;
 }
 
-static int trim_name(char **buffer, const char* prefix, const char* suffix)
+static void trim_name(char *buffer, const char* prefix, const char* suffix)
 {
-    char *pchr, *echr, *tmp;
-    int size;
-    if (NULL == *buffer) {
-        return 1;
+    char *pchr, *echr;
+    size_t buffer_len;
+
+    if (NULL == buffer) {
+        return;
     }
-    pchr = *buffer;
+
+    buffer_len = strlen (buffer);
+
+    pchr = buffer;
     if (NULL != prefix) {
-        pchr = strstr(*buffer, prefix);
-        if (NULL != pchr) {
-            pchr += strlen(prefix);
-        } else {
-            pchr = *buffer;
+        size_t prefix_len = strlen (prefix);
+
+        if (0 == strncmp (buffer, prefix, prefix_len)) {
+            pchr += prefix_len;
         }
     }
+
     /* trim spaces at the beginning */
-    while (' ' == *pchr || '\t' == *pchr) {
+    while (isspace (*pchr)) {
```

Comment:
I realize that this is the way it was before you fixed it -- but shouldn't the prefix check come *after* the beginning-of-string white space trim?

I ask because the suffix check comes *after* the end-of-string white space trim.